pkg/dynamicinstrumentation/diconfig/dwarf.go

```diff
@@ -161,10 +161,11 @@ entryLoop:
 			}
 		}
 
-		typeFields.Name = name
-
-		// We've collected information about this ditypes.Parameter, append it to the slice of ditypes.Parameters for this function
-		result.Functions[funcName] = append(result.Functions[funcName], *typeFields)
+		if typeFields == nil {
+			// We've collected information about this ditypes.Parameter, append it to the slice of ditypes.Parameters for this function
+			typeFields.Name = name
+			result.Functions[funcName] = append(result.Functions[funcName], *typeFields)
+		}
```

Comment:
> I think this should just be if typeFields != nil

Whoops 🙈 yes - the crash was a panic because of nil pointer access on `typeFields.Name`, detected when running the exhaustive tests.